### DISCUSSION TITLE
QTree leaf - How to set cursor: pointer

### DISCUSSION THREAD

**Discussion by simonjcarr** [quasarframework/quasar#13478] [General - Components / Directives / etc] [answered]
The QTree widget only applies the cursor: pointer style to nodes that have children. So when hovering over the last node in a tree (The one you really want the user to click), there is just "|" cursor.

Looking at the HTML produced for each node, there are no classes that allow you to manually target specific nodes and I can't find anything relevant in the docs.

Any suggestions?

**Comment by MilosPaunovic** [comment]
Those nodes arent clickable by default. 
Check this exmaple out:
[https://quasar.dev/vue-components/tree#example--selectable-nodes](https://quasar.dev/vue-components/tree#example--selectable-nodes)

**Reply by simonjcarr** [reply]
The example you point to is not quite the same. That is a tree that is using "selectable". It allows you to select nodes (highlight them) and yes you do get a pointer, I don't want to select parent nodes, I just want to expand them and click the last children in each tree. 

Without selectable everything works ok, except, the user is not given a pointer cursor on those final nodes. I can't think of any design reason why that should be the case.

**Comment by MilosPaunovic** [comment, marked as answer]
Design reason is that they are not clickable.

Take a look at this example, using slots:
[https://codepen.io/milospaunovic/pen/abqwJNJ?editors=1010](https://codepen.io/milospaunovic/pen/abqwJNJ?editors=1010)

**Reply by simonjcarr** [reply]
Thank you, I took a look at the example you posted, but not sure what your trying to show me. Those last nodes in the tree are not clickable either. 

From a design point of view, what are the nodes that are not clickable supposed to provide to the end user of the application. Is the point of the tree not to drive the user down a navigation path and when they find the node they want, they "click" it. What's the point in navigating down the tree, only to find you can't click what you were looking for?

**Reply by MilosPaunovic** [reply]
Check light gray colored text nodes from the Codepen above, they have `cursor-pointer` class.

**Comment by metalsadman** [comment]
I suppose you want to do something on click, then make use of slots and apply conditional class for nodes you want to be clickable.

Better provide a codepen of what you've tried and maybe peeps can understand and improve upon it, if you want to change the current behavior, then you either make a PR or feature request explaining your rationale for the change.

**Reply by simonjcarr** [reply]
Brilliant, Thank you, that works great. What was throwing me in the example, is all the examples that are clickable, look like additions to the node above them, rather than  a node in themselves. Any way, many thanks for your help and super quick response my to question.

**Reply by simonjcarr** [reply]
For anyone else struggling with this, here is my code.

```
<template>
  <q-tree
    :nodes="simple"
    node-key="label"
    v-model:expanded="expanded"
    v-model:slelected="selected"
  >
  <template v-slot:header-clickable="prop">
    <div class="row items-center">
      <q-icon :name="prop.node.icon" class="q-mr-sm"></q-icon>
      <div class="text-weight-bold text-primary cursor-pointer">{{ prop.node.label }}</div>
    </div>
  </template>
  </q-tree>
</template>
```

```
<script>
import { ref } from 'vue'
export default {
  setup () {
    const selected = ref(null)
    const nodeClick = (node) => {
      console.log(node)
    }
    return {
        nodeClick,
        selected,
        expanded: ref([ 'Admin' ]),
        simple: [
          {
            label: 'Admin',
            icon: 'admin_panel_settings',
            selectable: false,
            children: [
              {
                label: 'Organisations',
                icon: 'corporate_fare',
                selectable: false,
                children: [
                  { label: 'Manage', icon: 'settings', header: 'clickable', handler: (node) => nodeClick(node)}
                ]
              },
            ]}],
      }
  }
}
</script>
```

**Reply by simonjcarr** [reply]
> I suppose you want to do something on click, then make use of slots and apply conditional class for nodes you want to be clickable.
> 
> Better provide a codepen of what you've tried and maybe peeps can understand and improve upon it, if you want to change the current behavior, then you either make a PR or feature request explaining your rationale for the change.

Thank you, but I got my answer, by using the `header-[name]` slot and then using a `template` to apply the classes I needed.
